frontend/src/views/account/AddShop.vue

```diff
@@ -1,121 +1,185 @@
 <script setup lang="ts">
-import Header from '@/components/layout/Header.vue';
+import HeaderContainer from '@/components/layout/HeaderContainer.vue';
 import MainContainer from '@/components/layout/MainContainer.vue';
 import FormGroup from '@/components/layout/FormGroup.vue';
+import { useRouter } from 'vue-router';
 
 import { useAlertStore } from '@/stores/alert.store';
 import { useAuthStore } from '@/stores/auth.store';
 import { useShopStore } from '@/stores/shop.store';
 
-import { Form, Field } from 'vee-validate';
-import { useRouter } from 'vue-router';
-import * as Yup from 'yup';
-import {  RouterInput } from "@/helpers/trpc";
+import { useForm } from 'vee-validate';
+import { toTypedSchema } from '@vee-validate/zod';
+import * as z from 'zod';
 
 const authStore = useAuthStore();
 const shopStore = useShopStore();
 const alertStore = useAlertStore();
 const router = useRouter();
 
-const isValidUrl = (url: string) => {
-    try {
-        new URL(url);
-    } catch (e) {
-        return false;
-    }
-    return true;
-};
-
-const schema = Yup.object().shape({
-  name: Yup.string().required('Shop name is required'),
-  shopUrl: Yup.string().required('Shop URL is required').test('is-url-valid', 'Shop URL is not valid', (value) => isValidUrl(value)),
-  orgId: Yup.number().required('Organization is required'),
-  clientId: Yup.string().required('Client ID is required'),
-  clientSecret: Yup.string().required('Client Secret is required'),
+const { values, handleSubmit, errors, isSubmitting  } = useForm({
+    validationSchema: toTypedSchema(z.object({
+        name: z.string().min(1, 'Shop name is required'),
+        shopUrl: z.string().url('Shop URL is required'),
```

Comment:
does it work with http://localhost:8000 ? the special check before was to handle this